templates/gitlab/mender-deployment-create.gitlab-ci.yml

```diff
@@ -0,0 +1,50 @@
+# Use this template to create deployments for Mender artifacts from MENDER_ARTIFACTS_FOLDER folder.
+#
+# The template is designed to be used with the includes keyword or copied into an existing
+# configuration. It requires `MENDER_SERVER_URL` and `MENDER_SERVER_ACCESS_TOKEN` variables
+# to work and can be used to create a deployment on a Mender server.
+#
+# It's possible to set custom values by setting the following environment variables:
+# - `MENDER_DEPLOYMENT_GROUP`: The name of Mender devices group a deployment will be triggered to.
+#                              One of `MENDER_DEPLOYMENT_GROUP` or `MENDER_DEVICES_LIST` is required.
+#                              `MENDER_DEVICES_LIST` takes the priority if both are set.
+# - `MENDER_DEVICES_LIST`:     The list of Mender devices a deployment will be triggered to.
+# - `MENDER_DEPLOYMENT_NAME`:  The name of Mender deployment.
+# - `MENDER_RELEASE_NAME`:     The name of Mender release.
+
+include:
+  - project: 'Northern.tech/Mender/mender-ci-workflows'
+    file: 'templates/gitlab/common.gitlab-ci.yml'
+    ref: 'master'
+
+variables:
+  MENDER_DEPLOYMENT_GROUP: "" # The name of Mender devices group a deployment will be triggered to. One of `MENDER_DEPLOYMENT_GROUP` or `MENDER_DEVICES_LIST` is required. `MENDER_DEVICES_LIST` takes the priority if both are set
+  MENDER_DEVICES_LIST: ""     # The list of Mender devices a deployment will be triggered to
+  MENDER_RELEASE_NAME: ""     # The name of Mender release
+  MENDER_DEPLOYMENT_NAME: ""  # The name of Mender deployment
+
+.mender:create:deployment:
+  image: mendersoftware/mender-ci-tools:mender-master
```

Comment:
Similar to what we have done for `.mender:build:artifact`, we should have here a before_script meant to be overridden by the user

```
before_script:
    # to be overwritten after the job is inherited
    - echo "Please override before_script in your job defining MENDER_DEPLOYMENT_NAME"
    - exit 1
```

Reply:
Wait, see my other comment. We don't need this `before_script` protection here because we don't need the user to specify `before_script` section anyway.

Reply:
It's different ways of defining required by templates environment variables and depending from a use case and intention one of `variables:` or `before_script` can be used.

If, for instance, a `MENDER_DEPLOYMENT_NAME` is persistent then `variables:` is the way to go.

To generate unique names based on desired naming patterns,
someone might want to define those mandatory environment variables in other jobs and [pass them from another job](https://docs.gitlab.com/ee/ci/variables/#pass-an-environment-variable-to-another-job), or use `before_script` for that.

From my side it's intentional to show different ways of achieving the same, because it gives required in real use cases flexibility.

Reply:
My concern is about having a consistent and intuitive API. Sure users can do things different way, but we should try as much as we can to present a consistent API.

But okey, I leave it up to you. As long as we have the checks for the variables on the template jobs that should catch any issue of a user missing them from either `variables` or `before_script`.